test/objects_test.js

```diff
@@ -49,4 +49,39 @@ describe('objects', () => {
     assert.deepEqual(Page.fromRaw({data: 1, before: 2, after: 3}), new Page(1, 2, 3))
     assert.deepEqual(new Page([1, 2, 3], 2, 3).mapData(_ => _ + 1), new Page([2, 3, 4], 2, 3))
   })
+
+  it('time conversion', () => {
+    const dt = new Date()
+    assert.deepEqual(new FaunaTime(dt).date, dt)
+
+    const epoch = new Date(Date.UTC(1970, 0, 1))
+    const ft = new FaunaTime(epoch)
+    assert.deepEqual(ft, new FaunaTime('1970-01-01T00:00:00.000Z'))
+    assert.deepEqual(ft.date, epoch)
+  })
+
+  it('time', () => {
+      const test_ts = new FaunaTime('1970-01-01T00:00:00.123456789Z')
```

Comment:
Once we've parsed, say, a 4 hour offset, we have to add 4 hours to the time, which may mean incrementing the day, which may mean incrementing the month, etc... not something that can be done simply with string manipulation. The Date builtin handles all this math but only with millisecond precision. We could parse out the nanoseconds, then do Date arithmetic, then reintroduce nanos, but would that be worth the complexity? The ability to construct a FaunaTime out of a string is really only meant to be used by us when converting a JSON response.